src/sagas/delete-project.saga.js

```diff
@@ -95,29 +96,33 @@ export function* deleteProject({ project }: Action): Saga<void> {
   const nextSelectedProjectId = getNextProjectId(projects, project.id);
 
   if (shouldDeleteFromDisk) {
-    // Delete from disk tasks some time, so show a loading screen
-    yield put(startDeletingProject());
+    try {
+      // Delete from disk tasks some time, so show a loading screen
+      yield put(startDeletingProject());
+
+      // Run the deletion from disk
+      // first delete node_modules folder permanently (faster than moving to trash)
+      yield call(waitForAsyncRimraf, project.path);
+
+      // delete project folder
+      yield call([shell, shell.moveItemToTrash], project.path);
```

Comment:
@melanieseltzer  Sorry, for the delay. It's not working for my situation but I think you could add the following after the call of `moveItemToTrash`:
```js
     // Check if project folder is removed
      const exists = yield call([fs, fs.existsSync], project.path);
      if (exists) {
        throw new Error('deleting-failed');
      }
```
In the catch handler after the dialog a `loadDependencyInfoFromDisk(project.id, project.path)` needs to be dispatched as it's likely that the `node_modules` are now missing.

This will break the project but I'm working on a reinstall feature so it's also possible to get the dependencies back. The feature is not finished because I'd like to have a progress bar on the loading screen as installing takes pretty long.

Reply:
Hm I tried it out (and committed), but the suggested `exists` code won't run on my end. I think it's trying to delete and failing automatically due to me making it read only, so the catch block runs before the `exists` code gets a chance to (at least that's what I'm theorizing is happening).

Do you want to pull the branch and try it out with the Windows error, if you have some time?

I also don't see `yield put(loadDependencyInfoFromDisk(project.id, project.path));` working on my end, because for me node_modules doesn't get removed. But I think I can see it being necessary for other cases so I do think we need it.

Reply:
Sure, I'll have a look later today.
Yes, it's likely that your delete fails and it throws before the exists check. For my case it wasn't throwing and it failed silently.
I think that's OK as the user will get a message for both situations. I haven't tried read-only yet. But I'll also test this.